### PR TITLE
Adds a preference to switch walk hotkey toggle.

### PR DESCRIPTION
**Pull request by r1ks-iwnl**
## About The Pull Request
Changes the move intent hotkey to be a toggle instead of held down and makes it a preference inside the stats menu's preference tab.

## Why It's Good For The Game
In my opinion it's much easier to just tap alt once to switch between walking and running especially when having to block in combat which results in the player having to constantly hold down alt.

## Changelog
:cl:
add: Preference to switch the walk hotkey between held down and a toggle inside the stats menu's preference tab.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Should be an option, its really convenient to have a walk hold button especially for blocking. (Well it would be if the button wasn't alt which disables clicking on things)

**Comment by r1ks-iwnl** [comment]
Alright i could try doing that later

**Comment by r1ks-iwnl** [comment]
i tried making it a setting inside the keybinds menu right next to where the walk hotkey is first but that didn't go very well so i just did this instead

File: code/datums/keybinding/mob.dm [hunk 1]
```diff
@@ -145,9 +145,10 @@
 
 /datum/keybinding/mob/toggle_move_intent/up(client/user)
 	if(!user.mob) return
-	var/mob/M = user.mob
-	M.toggle_move_intent()
-	return TRUE
+	if(user.prefs.held_walk == TRUE)
```
Comment (PowerfulBacon) [inline code/datums/keybinding/mob.dm]:
```suggestion
	if(user.prefs.held_walk)
```

**Review by utilbird — APPROVED** [review]

**Pull request merged**